src/components/Stats.astro

```diff
@@ -0,0 +1,336 @@
+        <div class="stat-content">
+          <div
+            class="stat-icon"
+            style={stat.icon !== "image" ? `color: ${stat.color};` : ""}
+          >
+            {stat.icon === "cake" && (
+              <svg
+                width="48"
+                height="48"
+                viewBox="0 0 48 48"
+                fill="none"
+                xmlns="http://www.w3.org/2000/svg"
+              >
+                <path
+                  d="M24 12C26.2 12 28 10.2 28 8C28 7.24 27.8 6.54 27.42 5.94L24 0L20.58 5.94C20.2 6.54 20 7.24 20 8C20 10.2 21.8 12 24 12ZM36 18H26V14H22V18H12C8.68 18 6 20.68 6 24V42C6 43.1 6.9 44 8 44H40C41.1 44 42 43.1 42 42V24C42 20.68 39.32 18 36 18Z"
+                  fill="currentColor"
+                />
+              </svg>
+            )}
+            {stat.icon === "image" && (
+              <img
+                src={stat.imageSrc}
+                alt="Prize icon"
+                width="64"
+                height="64"
+              />
+            )}
+            {stat.icon === "star" && (
+              <svg
+                width="48"
+                height="48"
+                viewBox="0 0 48 48"
+                fill="none"
+                xmlns="http://www.w3.org/2000/svg"
+              >
+                <path
+                  d="M24 4L30.18 16.52L44 18.54L34 28.14L36.36 42.02L24 35.54L11.64 42.02L14 28.14L4 18.54L17.82 16.52L24 4Z"
+                  fill="currentColor"
+                />
+              </svg>
+            )}
+            {stat.icon === "medal" && (
+              <svg
+                width="48"
+                height="48"
+                viewBox="0 0 48 48"
+                fill="none"
+                xmlns="http://www.w3.org/2000/svg"
+              >
+                <path
+                  d="M24 4C16.26 4 10 10.26 10 18C10 28.5 24 44 24 44S38 28.5 38 18C38 10.26 31.74 4 24 4ZM14 18C14 12.48 18.48 8 24 8S34 12.48 34 18C34 23.76 28.24 32.38 24 37.76C19.76 32.38 14 23.7 14 18Z"
+                  fill="currentColor"
+                />
+              </svg>
+            )}
+            {stat.icon === "trophy" && (
+              <svg
+                width="48"
+                height="48"
+                viewBox="0 0 48 48"
+                fill="none"
+                xmlns="http://www.w3.org/2000/svg"
+              >
+                <path
+                  d="M38 10H34V6H14V10H10C7.8 10 6 11.8 6 14V16C6 21.1 9.84 25.26 14.78 25.88C16.04 28.88 18.74 31.14 22 31.8V38H14V42H34V38H26V31.8C29.26 31.14 31.96 28.88 33.22 25.88C38.16 25.26 42 21.1 42 16V14C42 11.8 40.2 10 38 10ZM10 16V14H14V21.64C11.68 20.8 10 18.6 10 16ZM38 16C38 18.6 36.32 20.8 34 21.64V14H38V16Z"
+                  fill="currentColor"
+                />
+              </svg>
+            )}
+          </div>
+          <div class="stat-info">
+            <div class="title-row">
+              <h3>{stat.title}</h3>
+              <div class="toggle-arrow">▶</div>
```

Comment:
we should probably use a unicode character or something here, this shows up as an emoji on mobile (at least on iOS)